### ISSUE TITLE
Could not find the entity manager

### ISSUE THREAD

**Issue by fritzmg** [terminal42/contao-shortlink#25]
Unfortunately there is still another error in `1.3.1`/`dev-main` under Contao **4.13**:

```
LogicException:
Could not find the entity manager for class "Terminal42\ShortlinkBundle\Entity\Shortlink". Check your Doctrine configuration to make sure it is configured to load this entity’s metadata.

  at vendor\doctrine\doctrine-bundle\Repository\LazyServiceEntityRepository.php:74
  at Doctrine\Bundle\DoctrineBundle\Repository\LazyServiceEntityRepository->initialize()
     (vendor\doctrine\doctrine-bundle\Repository\LazyServiceEntityRepository.php:49)
  at Doctrine\Bundle\DoctrineBundle\Repository\LazyServiceEntityRepository->__get('_em')
     (vendor\doctrine\orm\src\EntityRepository.php:310)
  at Doctrine\ORM\EntityRepository->getEntityManager()
     (vendor\terminal42\contao-shortlink\src\Repository\ShortlinkRepository.php:31)
  at Terminal42\ShortlinkBundle\Repository\ShortlinkRepository->findRouteCandidatesByAlias('')
     (vendor\terminal42\contao-shortlink\src\Routing\RouteProvider.php:35)
  at Terminal42\ShortlinkBundle\Routing\RouteProvider->getRouteCollectionForRequest(object(Request))
     (vendor\symfony-cmf\routing\src\NestedMatcher\NestedMatcher.php:140)
  at Symfony\Cmf\Component\Routing\NestedMatcher\NestedMatcher->matchRequest(object(Request))
     (vendor\symfony-cmf\routing\src\DynamicRouter.php:272)
  at Symfony\Cmf\Component\Routing\DynamicRouter->matchRequest(object(Request))
     (vendor\symfony-cmf\routing\src\ChainRouter.php:188)
  at Symfony\Cmf\Component\Routing\ChainRouter->doMatch('/', object(Request))
     (vendor\symfony-cmf\routing\src\ChainRouter.php:158)
  at Symfony\Cmf\Component\Routing\ChainRouter->matchRequest(object(Request))
     (vendor\symfony\http-kernel\EventListener\RouterListener.php:111)
  at Symfony\Component\HttpKernel\EventListener\RouterListener->onKernelRequest(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
     (vendor\symfony\event-dispatcher\Debug\WrappedListener.php:118)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
     (vendor\symfony\event-dispatcher\EventDispatcher.php:230)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.request', object(RequestEvent))
     (vendor\symfony\event-dispatcher\EventDispatcher.php:59)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (vendor\symfony\event-dispatcher\Debug\TraceableEventDispatcher.php:154)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (vendor\symfony\http-kernel\HttpKernel.php:139)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor\symfony\http-kernel\HttpKernel.php:75)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor\symfony\http-kernel\Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public\index.php:44)                
```

When running `debug:config doctrine` you can see that the automatically mapped entity is set up to use annotations, instead of attributes:

```yaml
doctrine:
    orm:
        entity_managers:
            default:
                mappings:
                    Terminal42ShortlinkBundle:
                        is_bundle: true
                        type: annotation
                        dir: src/Entity
                        mapping: true
```

Not sure how to fix this. I think this happens because in `symfony/doctrine-bridge` `5.x` (Contao 4.13) the default metadata type for automatically mapped entities is `annotation` while in `symfony/doctrine-bridge` `6.x` (Contao 5) the default is `attribute`.

`terminal42/contao-shortlink` probably needs to override the `doctrine.orm` config to explicitly set the entity mapping like so: https://symfony.com/doc/5.x/reference/configuration/doctrine.html#custom-mapping-entities-in-a-bundle

**Comment by aschempp** [comment]
any idea how this did work before in 4.13?

**Comment by fritzmg** [comment]
https://github.com/terminal42/contao-shortlink/pull/23 changed the Annotations to Attributes, as that is the default now in Symfony 6.

**Comment by fritzmg** [comment]
@FrischBlau for now you need to add a conflict for `1.3.0 || 1.3.1` for this package and install `^1.2` in your Contao 4.13 instance.

**Comment by FrischBlau** [comment]
Thanks for the quick reply!

**Comment by aschempp** [comment]
Looks like we configure this ourselves… should be fixed in eec356f952736644ee7ddaa311c00d0f5159fc6b (v1.3.2)
